CONTRIBUTING

```diff
@@ -1,282 +1,113 @@
-changelog entry so that the next reviewer better understands what is going on.
-
-Be sure to tell the reviewers what changes you are making and to thank them for
-their time.  Code review is a tiring and time-consuming  process, and reviewers
-sometimes get grumpy.  Even in that case, though, respond politely and address
-the problems they have pointed out.
-
-It is important to understand that patches must sometimes go through several
-iterations of review before being approved.  Not all patches will be accepted
-either.  It is important that you do not take any criticism personally.  Patch
-review is an objective and rational critique only of the content of the patch
-itself.
-
-B) Don't get discouraged - or impatient.
-After you have submitted your change, be patient and wait.  Reviewers are busy
-people and may not get to your patch right away.
+  ```
+    Related: rhbz#1234567
+  ```
 
-You should receive comments within a week or so; if that does not happen, make
-sure that you have sent your patches to the right place.  Wait for a  minimum of
-one week before resubmitting or pinging reviewers - possibly  longer during busy
-times like Fedora release stabilization.
+Use `Resolves` if the patch fixes the core issue which caused the bug.
+Use `Related` if the patch fixes an ancillary issue that is related to, but
+might not actually fix the bug.
 
-Note that there is a minimum review period of 24 hours for any patch.  The
-purpose of this rule is to ensure that all interested parties have an
-opportunity to review every patch.  When posting a patch before or after a
-holiday break it is important to extend this period as appropriate.
+__NOTE__: If you're using `Resolves` or `Related` you should omit the bug id
+reference from the first line of the commit message.
+
+#### Creating a Pull Request
+When creating a pull request for blivet you have to give some thought
+to what branch to base your pull request on. You should choose the
+branch corresponding to the oldest `x.y` release that should include
+your work. If we are currently working toward a `2.1.0` release and
+your pull request is a bug fix, you'll want to apply it at least as
+far back as `2.0-devel`.
+
+Note that there is a minimum review period of 24 hours for any patch.
+The purpose of this rule is to ensure that all interested parties have
+an opportunity to review every patch.  When posting a patch before or
+after a holiday break it is important to extend this period as
+appropriate.
+
+#### Merging a Pull Request
+After hitting the big green button, you will also need to merge your
+work to other active development branches and to `master`.
+
+- if your work is on `x.y-devel`, merge it onto `x.{y+1}-devel`
+      ```
+      git checkout 2.1-devel
+      ```
+      ```
+      git merge --no-ff 2.0-devel
+      ```
+      Take a look to make sure things are as expected, then...
+      ```
+      git push
```

Comment:
Does this actually work? Maybe I have some different config, but just ``git push`` pushes only the current branch to its upstream. What I have to do is something like this ``git push origin master 2.0-devel 2.1-devel 3.0-devel``.